skcosmo/metrics/_reconstruction_measures.py

```diff
@@ -0,0 +1,353 @@
+from skcosmo.model_selection import train_test_split
+
+from ..preprocessing import NormalizeScaler
+from ..linear_model import OrthogonalRegression, RidgeRegression2FoldCV
+
+import numpy as np
+
+from joblib import Parallel, delayed
+
+
+def pointwise_global_reconstruction_error(
+    X,
+    Y,
+    test_size=None,
+    train_size=None,
+    random_state=None,
+    shuffle=True,
+    train_test_overlap=False,
+    scaler=None,
+    estimator=None,
+):
+    """Computes the pointwise global reconstruction error fitting the X dat
+    to reconstruct the Y data.
+
+    If used with X and Y of shape (samples, features) it computes the pointwise
+    global reconstruction error of the features as defined in Ref. [frm]_.
+    If used with X and Y of shape(features, samples) it computes the
+    reconstruction error of the samples. The default parameters mimics the
+    ones of Ref. [frm]_.
+
+    Parameters
+    ----------
+    X : array_like of shape (n_samples, X_n_features)
+        For feature reconstruction use input shape (samples, features).
+        For sample reconstruction use input shape (features, samples).
+
+    Y : array_like of shape (n_samples, Y_n_targets)
+        For feature reconstruction use input shape (samples, features).
+        For sample reconstruction use input shape (features, samples).
+
+    test_size : float or int, default=None
+        If float, should be between 0.0 and 1.0 and represent the proportion
+        of the dataset to include in the test split. If int, represents the
+        absolute number of test samples. If None, the value is set to the
+        complement of the train size. If ``train_size`` is also None, it will
+        be set to 0.5.
+
+    train_size : float or int, default=None
```

Comment:
what will happen if train_size is not None but test_size is?

Reply:
Is something missing in the doc? For me it reads like the complement is taken. This is basically a copy paste from sklearn
```py
    test_size : float or int, default=None
        If float, should be between 0.0 and 1.0 and represent the proportion
        of the dataset to include in the test split. If int, represents the
        absolute number of test samples. If None, the value is set to the
        complement of the train size. If ``train_size`` is also None, it will
        be set to 0.5.

    train_size : float or int, default=None
        If float, should be between 0.0 and 1.0 and represent the
        proportion of the dataset to include in the train split. If
        int, represents the absolute number of train samples. If None,
        the value is automatically set to the complement of the test size.
```

Reply:
I think I'm just missing a bit of understanding about parameter dissonance. If train_size, test_size, and train_test_overlap are at odd (for instance 0.7, 0.7, False), will it throw an error or will one set of variables override the others?

Reply:
so `train_test_overlap=False` gives you sklearn behavior, which is an error because train and test set may not overlap (train size + test size <= overall size). The error in this case is also verbose and explains what is wrong.